mapie/multi_label_classification.py

```diff
@@ -0,0 +1,795 @@
+        calib_size: Optional[float] = .3,
+    ) -> MapieMultiLabelClassifier:
+        """
+        Fit the base estimator or use the fitted base estimator on
+        batch data. All the computed risks will be concatenated each
+        time the partial_fit method is called.
+
+        Parameters
+        ----------
+        X : ArrayLike of shape (n_samples, n_features)
+            Training data.
+
+        y : NDArray of shape (n_samples, n_classes)
+            Training labels.
+
+        calib_size: Optional[float]
+            Size of the calibration dataset with respect to X if the
+            given model is None and that MAPIE need to fit a
+            LogisticRegression.
+
+            By default .3
+
+        _refit: bool
+            Whether or not refit MAPIE from scratch.
+
+            By default False
+
+        Returns
+        -------
+        MapieMultiLabelClassifier
+            The model itself.
+        """
+        # Checks
+        first_call = self._check_partial_fit_first_call()
+        self._check_parameters()
+
+        X, y = indexable(X, y)
+        _check_y(y, multi_output=True)
+        estimator, X, y = self._check_estimator(
+            X, y, self.estimator,
+            _refit, calib_size
+        )
+
+        y = cast(NDArray, y)
+        X = cast(NDArray, X)
+
+        self._check_all_labelled(y)
+        self.n_samples_ = _num_samples(X)
+
+        # Work
+        if first_call or _refit:
+            self.single_estimator_ = estimator
+            y_pred_proba = self.single_estimator_.predict_proba(X)
+            y_pred_proba_array = self._transform_pred_proba(y_pred_proba)
+            self.theta_ = np.zeros(X.shape[1])
+            self.risks = self._compute_risks(y_pred_proba_array, y)
+        else:
+            if X.shape[1] != self.theta_.shape[0]:
+                msg = "Number of features %d does not match previous data %d."
+                raise ValueError(msg % (X.shape[1], self.theta_.shape[0]))
+            self.single_estimator_ = estimator
+            y_pred_proba = self.single_estimator_.predict_proba(X)
+            y_pred_proba_array = self._transform_pred_proba(y_pred_proba)
+            partial_risk = self._compute_risks(y_pred_proba_array, y)
+            self.risks = np.concatenate([self.risks, partial_risk], axis=0)
+
+        return self
+
+    def fit(
+        self,
+        X: ArrayLike,
+        y: ArrayLike,
+        calib_size: Optional[float] = .3
+    ) -> MapieMultiLabelClassifier:
+        """
+        Fit the base estimator or use the fitted base estimator.
+
+        Parameters
+        ----------
+        X : ArrayLike of shape (n_samples, n_features)
+            Training data.
+
+        y : NDArray of shape (n_samples, n_classes)
+            Training labels.
+
+        calib_size: Optional[float]
+            Size of the calibration dataset with respect to X if the
+            given model is None and that MAPIE need to fit a
```

Comment:
None with ``
MAPIE need to fit -> fitting a logisticregression is needed (MAPIE is not a class)

Reply:
MAPIE does need anythging

Reply:
done